src/TranslatableTrait.php

```diff
@@ -15,109 +18,104 @@ trait TranslatableTrait
+     * Translates the given message.
+     *
+     * When a number is provided as a parameter named "%count%", the message is parsed for plural
+     * forms and a translation is chosen according to this number using the following rules:
+     *
+     * Given a message with different plural translations separated by a
+     * pipe (|), this method returns the correct portion of the message based
+     * on the given number, locale and the pluralization rules in the message
+     * itself.
+     *
+     * The message supports two different types of pluralization rules:
+     *
+     * interval: {0} There are no apples|{1} There is one apple|]1,Inf] There are %count% apples
+     * indexed:  There is one apple|There are %count% apples
      *
-     * _m = multi
+     * The indexed solution can also contain labels (e.g. one: There is one apple).
+     * This is purely for making the translations more clear - it does not
+     * affect the functionality.
      *
-     * @param string        $string     a string to be translated with sprintf occurence
-     * @param array<array>  $args       an array of array containing args to be translated using methods _ and _d
+     * The two methods can also be mixed:
+     *     {0} There are no apples|one: There is one apple|more: There are %count% apples
      *
-     * @return string
+     * An interval can represent a finite set of numbers:
+     *  {1,2,3,4}
+     *
+     * An interval can represent numbers between two numbers:
+     *  [1, +Inf]
+     *  ]-1,2[
+     *
+     * The left delimiter can be [ (inclusive) or ] (exclusive).
+     * The right delimiter can be [ (exclusive) or ] (inclusive).
+     * Beside numbers, you can use -Inf and +Inf for the infinite.
+     *
+     * @see https://en.wikipedia.org/wiki/ISO_31-11
+     *
+     * @param string      $id         The message id (may also be an object that can be cast to string)
+     * @param array       $parameters An array of parameters for the message
+     * @param string|null $domain     The domain for the message or null to use the default
+     * @param string|null $locale     The locale or null to use the default
+     *
+     * @throws \InvalidArgumentException If the locale contains invalid characters
+     *
+     * @return string The translated string
      */
-    public function _m(string $string, array $args)
+    public function _($id, array $parameters = [], $domain = null, $locale = null) :string
     {
-        $translated_args = [];
-        $translated_args[] = $this->_($string);
-        $this->_processTranslatedArray($translated_args, $args);
+        // case if is child Object which use AppScopeTrait + TranslatableTrait
+        // that was already add to an Object which use AppScopeTrait + ContainerTrait + TranslatableTrait
+        if ($this->hasTranslatorInAppScope()) {
+            return $this->app->_($id, $parameters, $domain, $locale);
+        }
+
+        // case if use TranslatableTrait and translator is defined
```

Comment:
I don't think we should even implement this. It's overkill. 

Most of ATK UI will create 'app' if missing.

Reply:
1. translator can be removed, i understand your point and your statement the other day about not using property in traits, it became a dynamic property.
2. The idea was to use TranslatableTrait : 
- App which use TranslatableTrait, direclty calling method $this->_ 
- Objects add via add which use TranslatableTrait => calls $this->_ and delegate to $this->app->_
- Object that implements only TranslatableTrait => calling $this->_ 

if $translator is not an instance of TranslatorInterface, it will fallback to Symfony implementation that don't have any translation to be loaded, it just process the $id with $parameters and return a correct string with excepted substituition as intended in Symfony.

about this code : 
```php
$translator = new class() implements TranslatorInterface {
        use TranslatorTrait;
};
```
Initally i defined as a separate object and at that place there was something like : 
```php
if(!$this->translator) {
        $this->translator = new FallbackTranslator();
}
```
thinking about wrongly bypass the second condition. I think that can be better, because if we set a translator with SetTranslator, condition will stop before a check for is own proprerty $translator.
if no AppScope